### DISCUSSION TITLE
Qt6.5 upgrade - facebook videos can't be played

### DISCUSSION THREAD

**Discussion by gadefox** [qutebrowser/qutebrowser#7905] [General]
This issue is probably not a bug, therefore I'm writing here.. I was happy with `Qt5`, but unfortunately the `WebEngine` is outdated and I was getting the silly messages about unsupported web browser ([discord.gnome.org](https://discourse.gnome.org) etc).

I spent whole day with the `Qt6.4.2` (`debian` packages) - causes the crash, no error messages in logs.. `Qt6.5` works fine (installed using `mkvenv`) except the videos in `facebook`. When I click the big white arrow, the frame turns black and nothing happens. Any idea what could be the problem? Ta

[qute://version](https://bpa.st/AE7QW)

The same behavior with `--temp-basedir` parameter..

**Comment by The-Compiler** [comment]
Based on `Qt library executable path: /usr/local/lib/qt6/libexec, data path: /usr/local/lib/qt6` in your vversion output, your Qt is neither installed via Debian packages, nor via `mkvenv.py`. Most likely, this is a Qt build without proprietary codec support enabled.

**Comment by gadefox** [comment]
Nah, it's installed using `mkvenv.py` and I copied the libraries from `~` to `/usr/local`.

**Comment by gadefox** [comment]
Okay, I reinstalled the `qutebrowser` via `mkvenv` (`./scripts/mkvenv.py --skip-docs --pyqt-version 6.5 --pyqt-type=binary
`)

```
Python: 3.11.5 (main, Aug 29 2023, 15:31:31) [GCC 13.2.0]
qVersion: 6.5.2
QT_VERSION_STR: 6.5.2
PYQT_VERSION_STR: 6.5.2
Qt seems to work properly!
```

I'll check the facebook, but I need to close this instance..

**Comment by gadefox** [comment]
[venv::qute://version](https://bpa.st/K6EAQ)

The same problem.. I'll try to use `Qt6.4` instead.

**Reply by The-Compiler** [reply]
It won't work with a Qt installed via `mkvenv.py`, as that comes without proprietary codec support.

**Reply by gadefox** [reply]
I see.. Well, it's weird, cause the `qutebrowser` with `debian` packages (`Qt6.4.2`) can't initialize `OpenGL` (see bellow) although the `Qt6.4` installed via `mkvenv` works fine.

testbrowser_webengine.py:
```
failed to create drawable
qt.webenginecontext:

GL Type: desktop
Surface Type: OpenGL
Surface Profile: CompatibilityProfile
Surface Version: 4.5
QSG RHI Backend: OpenGL
Using Supported QSG Backend: yes
Using Software Dynamic GL: no
Using Multithreaded OpenGL: yes

Init Parameters:
  *  application-name testbrowser_webengine.py
  *  browser-subprocess-path /usr/lib/qt6/libexec/QtWebEngineProcess
  *  create-default-gl-context
  *  disable-features ConsolidatedMovementXY,InstalledApp,BackgroundFetch,WebOTP,WebPayments,WebUSB,PictureInPicture
  *  disable-setuid-sandbox
  *  disable-speech-api
  *  enable-features NetworkServiceInProcess,TracingServiceInProcess
  *  enable-threaded-compositing
  *  in-process-gpu
  *  use-gl desktop

failed to create drawable
[119236:119262:0910/010559.213439:ERROR:gl_surface_glx_qt.cpp(157)] glXCreatePbuffer failed.
[119236:119262:0910/010559.213573:ERROR:gpu_info_collector.cc(80)] gl::GLContext::CreateOffscreenGLSurface failed
[119236:119262:0910/010559.213689:ERROR:gpu_info_collector.cc(342)] Could not create surface for info collection.
failed to create drawable
[119236:119262:0910/010559.233117:ERROR:gl_surface_glx_qt.cpp(157)] glXCreatePbuffer failed.
failed to create drawable
[119236:119262:0910/010559.244785:ERROR:gl_surface_glx_qt.cpp(157)] glXCreatePbuffer failed.
[119236:119262:0910/010559.244934:ERROR:gpu_info_collector.cc(80)] gl::GLContext::CreateOffscreenGLSurface failed
[119236:119262:0910/010559.245034:ERROR:gpu_info_collector.cc(342)] Could not create surface for info collection.
failed to create drawable
failed to create drawable
Loading started
Loading progress: 0%
```

opengl_info.py:
```
failed to create drawable
GLES: False
Vendor: Mesa
Renderer: llvmpipe (LLVM 15.0.7, 128 bits)
Version: 4.5 (Compatibility Profile) Mesa 23.1.6-1
Shading language version: 4.50
```

**Reply by gadefox** [reply]
The same scripts launched in `venv`:

opengl_info.py:
```
GLES: False
Vendor: NVIDIA Corporation
Renderer: GeForce GT 240M/PCIe/SSE2
Version: 3.3.0 NVIDIA 340.108
Shading language version: 3.30 NVIDIA via Cg compiler
```

testbrowser_webengine.py:
```
qt.webenginecontext:

GL Type: desktop
Surface Type: OpenGL
Surface Profile: CompatibilityProfile
Surface Version: 3.3
QSG RHI Backend: OpenGL
Using Supported QSG Backend: yes
Using Software Dynamic GL: no
Using Multithreaded OpenGL: yes

Init Parameters:
  *  application-name testbrowser_webengine.py
  *  browser-subprocess-path /home/fox/.build/qutebrowser-main/.venv2/lib/python3.11/si
  *  create-default-gl-context
  *  disable-features ConsolidatedMovementXY,InstalledApp,BackgroundFetch,WebOTP,WebPay
  *  disable-setuid-sandbox
  *  disable-speech-api
  *  enable-features NetworkServiceInProcess,TracingServiceInProcess
  *  enable-threaded-compositing
  *  in-process-gpu
  *  use-gl desktop

Loading started
Loading progress: 0%
Loading progress: 22%
Loading progress: 70%
Loading progress: 100%
Loading finished
```

**Reply by The-Compiler** [reply]
No idea why that is... Either Debian's patching (or way of building things) breaks something, or there was some kind of bugfix that is in 6.4.3 but was not in 6.4.2.
